tests/unittests/OnnxImporterTest.cpp

```diff
@@ -5437,3 +5437,143 @@ TEST_F(OnnxImporterTest, softmax13) {
+  auto result = bindings.get(output)->getHandle();
+  std::vector<dim_t> expectedDims = {1, 16, 26, 26};
+  EXPECT_TRUE(result.dims().vec() == expectedDims);
+
+  // This Model has three operators
+  // Here Conv and ReLu are specified in the yaml file to run in fp16.
+  //  Input: Placeholder(float)
+  //      |
+  //      V
+  //  Conv (set to float16)
+  //      |
+  //      V
+  //  BatchNorm(float)
+  //      |
+  //      V
+  //  ReLu (set to float16)
+  //      |
+  //      V
+  //  Output: Placeholder(float)
+  // Check in the Glow graph the corresponding nodes are in FP16.
+  EXPECT_EQ(countNodeKind(F, Kinded::Kind::ConvertToNodeKind), 2);
+  for (auto &node : F->getNodes()) {
+    if (auto *MN = llvm::dyn_cast<MaxNode>(&node)) {
+      EXPECT_EQ(MN->getResult().getElementType(), ElemKind::Float16Ty);
+    } else if (auto *CN = llvm::dyn_cast<ConvolutionNode>(&node)) {
+      EXPECT_EQ(CN->getResult().getElementType(), ElemKind::Float16Ty);
+      auto *TN = llvm::dyn_cast<TransposeNode>(CN->getInput());
+      EXPECT_NE(nullptr, TN);
+      auto *CT = llvm::dyn_cast<ConvertToNode>(TN->getInput());
+      EXPECT_NE(nullptr, CT);
+    } else if (auto *CT = llvm::dyn_cast<ConvertToNode>(&node)) {
+      if (CT->getResult().getElementType() == ElemKind::FloatTy) {
+        auto *MN = llvm::dyn_cast<MaxNode>(CT->getInput());
+        EXPECT_NE(nullptr, MN);
+      }
+    }
+  }
+}
+
+TEST_F(OnnxImporterTest, mixedPrecisionConfigMultipleUsers) {
+  // This test checks when a list of node instances set to run in fp16
+  // has multiple users among which one user is not specified to run in fp16
+  ExecutionEngine EE;
+  auto &mod = EE.getModule();
+  std::string netFilename(
+      GLOW_DATA_PATH "tests/models/onnxModels/mixedPrecisionMultUsers.onnxtxt");
+  PlaceholderBindings bindings;
+  auto *F = mod.createFunction("main");
+  Placeholder *output;
+  Tensor data(ElemKind::FloatTy, {1, 3, 28, 28});
+
+  // Precision config info used in model loader
+  std::string precisionConfigFile(
+      GLOW_DATA_PATH "tests/models/onnxModels/mixedPrecisionMultUsers.yaml");
+  setModelLoaderPrecisionOpt(precisionConfigFile);
+
+  {
+    ONNXModelLoader onnxLD(netFilename, {"input"}, {&data.getType()}, *F);
+    output = EXIT_ON_ERR(onnxLD.getSingleOutput());
+    EXPECT_EQ(mod.getPlaceholders().size(), 2);
+    bindings.allocate(mod.getPlaceholders());
+  }
+  EE.compile(CompilationMode::Infer);
+  EE.run(bindings);
+  setConstantFoldLoaderOpsFlag(false);
+
+  auto result = bindings.get(output)->getHandle();
+  std::vector<dim_t> expectedDims = {1, 16, 26, 26};
+  EXPECT_TRUE(result.dims().vec() == expectedDims);
+
+  // This Model has five operators
+  // Here ReLu and BatchNorm appearing followed by ReLu
+  // are specified in the yaml file to run in fp16.
+  // ReLu node has two users and one among them is set to fp16
+  //        Input: Placeholder(float)
+  //                    |
+  //                    V
+  //              Conv (float)
```

Comment:
nit: Can you call this e.g. Convert? Conv is usually short for "Convolution" so it's confusing here

Reply:
Done